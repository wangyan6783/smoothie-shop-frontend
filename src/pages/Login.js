import React, { useState } from "react";
import { Link } from "react-router-dom";

function Login() {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");

  function handleUsername(e) {
    setUsername(e.target.value);
  }

  function handlePassword(e) {
    setPassword(e.target.value);
  }

  function handleSubmit(e) {
    e.preventDefault();
    fetch("http://localhost:5000/users/login", {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json"
      },
      body: JSON.stringify({
        username,
        password
      })
    })
      .then(r => r.json())
      .then(data => console.log(data));
  }

  return (
    <div className="login">
      <form className="form" onSubmit={handleSubmit}>
        <h1 className="heading-1 mb-md">Log in</h1>
        <div>
          <label className="form-text" htmlFor="username">
            Username
          </label>
          <input
            className="form-input"
            type="text"
            name="username"
            onChange={handleUsername}
          ></input>
        </div>
        <div className="mb-sm">
          <label className="form-text" htmlFor="password">
            Password
          </label>
          <input
            className="form-input mb-sm"
            type="text"
            name="password"
            onChange={handlePassword}
          ></input>
        </div>
        <button className="btn form-btn mb-sm" type="submit">
          Log in
        </button>
        <p className="form-text">
          Don't have an account?{" "}
          <Link className="form-link" to="/signup">
            Sign Up
          </Link>
        </p>
      </form>
    </div>
  );
}

export default Login;
