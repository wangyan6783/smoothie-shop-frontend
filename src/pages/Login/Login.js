import React, { useState } from "react";
import { Link, withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { setCurrentUser } from "../../redux/user/user.actions";

function Login(props) {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [loginError, setLoginError] = useState(false);

  function handleUsername(e) {
    setUsername(e.target.value);
  }

  function handlePassword(e) {
    setPassword(e.target.value);
  }

  function handleSubmit(e) {
    e.preventDefault();
    setUsername("");
    setPassword("");
    fetch("http://localhost:5000/users/login", {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json"
      },
      body: JSON.stringify({
        username,
        password
      })
    })
      .then(r => r.json())
      .then(data => {
        console.log(data);
        if (data === "error") {
          setLoginError(true);
        } else {
          props.setCurrentUser({ username: data.username });
          props.history.push("/");
        }
      });
  }

  return (
    <div className="login">
      <form className="form" onSubmit={handleSubmit}>
        <h1 className="heading-1 mb-md">Log in</h1>
        {loginError ? (
          <div className="form-error">Invalid username or password</div>
        ) : null}
        <div>
          <label className="form-text" htmlFor="login-username">
            Username
          </label>
          <input
            id="login-username"
            className="form-input"
            type="text"
            name="username"
            value={username}
            onChange={handleUsername}
          ></input>
        </div>
        <div className="mb-sm">
          <label className="form-text" htmlFor="login-password">
            Password
          </label>
          <input
            id="login-password"
            className="form-input mb-sm"
            type="password"
            name="password"
            value={password}
            onChange={handlePassword}
          ></input>
        </div>
        <button className="btn form-btn mb-sm" type="submit">
          Log in
        </button>
        <p className="form-text">
          {`Don't have an account? `}
          <Link className="form-link" to="/signup">
            Sign Up
          </Link>
        </p>
      </form>
    </div>
  );
}

const mapDispatchToProps = dispatch => ({
  setCurrentUser: user => dispatch(setCurrentUser(user))
});

export default withRouter(connect(null, mapDispatchToProps)(Login));
