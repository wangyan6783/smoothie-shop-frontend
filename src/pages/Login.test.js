import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Login from "./Login";

describe("Login Component", () => {
  it("render correctly", () => {
    const loginComponent = renderer
      .create(
        <Router>
          <Login />
        </Router>
      )
      .toJSON();
    expect(loginComponent).toMatchSnapshot();
  });
});
