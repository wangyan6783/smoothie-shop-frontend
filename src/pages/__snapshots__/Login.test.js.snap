// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component render correctly 1`] = `
<div
  className="login"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <h1
      className="heading-1 mb-md"
    >
      Log in
    </h1>
    <div>
      <label
        className="form-text"
        htmlFor="username"
      >
        Username
      </label>
      <input
        className="form-input"
        name="username"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="mb-sm"
    >
      <label
        className="form-text"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-input mb-sm"
        name="password"
        onChange={[Function]}
        type="text"
      />
    </div>
    <button
      className="btn form-btn mb-sm"
      type="submit"
    >
      Log in
    </button>
    <p
      className="form-text"
    >
      Don't have an account?
       
      <a
        className="form-link"
        href="/signup"
        onClick={[Function]}
      >
        Sign Up
      </a>
    </p>
  </form>
</div>
`;
